core/templates/domain/feature_gating/PlatformParameterFilterObjectFactory.ts

```diff
@@ -0,0 +1,174 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating PlatformParameterFilter domain objects.
+ */
+
+import cloneDeep from 'lodash/cloneDeep';
+
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+
+const constants = require('constants.ts');
+
+
+export enum PlatformParameterFilterType {
+  ServerMode = 'server_mode',
+  UserLocale = 'user_locale',
+  ClientType = 'client_type',
+  BrowserType = 'browser_type',
+  AppVersion = 'app_version',
+}
+
+export enum ServerMode {
+  Dev = 'dev',
+  Test = 'test',
+  Prod = 'prod',
+}
+
+export const ALLOWED_SERVER_MODE = [
+  ServerMode.Dev, ServerMode.Test, ServerMode.Prod].map(e => e.toString());
+
+export const SUPPORTED_SITE_LANGUAGE_IDS = (
+  constants.SUPPORTED_SITE_LANGUAGES.map((lang: {id: string}) => lang.id)
+);
+
+export const ALLOWED_CLIENT_TYPE = ['Android', 'Web'];
+
+export const ALLOWED_BROWSER_TYPE = [
+  'Chrome', 'Edge', 'Safari', 'Firefox', 'Others'];
+
+export const APP_VERSION_REGEXP = /^\d+(?:\.\d+)*$/;
+
+export const ALLOWED_OP_FOR_TYPE = {
+  [PlatformParameterFilterType.ServerMode]: ['='],
+  [PlatformParameterFilterType.UserLocale]: ['='],
+  [PlatformParameterFilterType.ClientType]: ['='],
+  [PlatformParameterFilterType.BrowserType]: ['='],
+  [PlatformParameterFilterType.AppVersion]: ['=', '<', '>', '<=', '>='],
```

Comment:
Are this & the regex for app version up-to-date? App flavors seem to be missing--please add.